<?xml version="1.0"?>

<launch>
  <arg name="ns" default="espros_cam660"/>
  <arg name="nn" default="camera"/>
  <arg name="respawn" default="false"/>  
  <arg name="lens_data" default="lens_wf"/>
  <arg name="lens_center_offset_x" default="0"/>
  <arg name="lens_center_offset_y" default="0"/>
  <arg name="image_type" default="1"/>
  <arg name="start_stream" default="false"/>
  <arg name="trigger_single_shot" default="false"/>
  <arg name="hdr_mode" default="0"/>
  <arg name="integration_time_tof_1"  default="50"/>
  <arg name="integration_time_tof_2"  default="400"/>
  <arg name="integration_time_tof_3"  default="0"/>
  <arg name="integration_time_gray" default="3000"/>
  <arg name="offset" default="0"/> 
  <arg name="roi_left_x" default="0"/>
  <arg name="roi_right_x" default="319"/>
  <arg name="roi_height" default="240"/>
  <arg name="min_amplitude" default="100"/>
  <arg name="min_distance" default="0"/>
  <arg name="max_distance" default="6000"/>
  <arg name="frequence_modulation" default="0"/>
  <arg name="channel" default="0"/>
  <arg name="median_filter" default="false"/>
  <arg name="average_filter" default="false"/>
  <arg name="edge_filter_threshold" default="0"/>
  <arg name="temporal_filter_factor" default="0.0"/>
  <arg name="temporal_filter_threshold" default="0"/>
  <arg name="interference_detect_limit" default="0"/>
  <arg name="use_last_value" default="false"/>
  <arg name="cartesian" default="true"/>
  <arg name="publish_images" default="true"/>
  <arg name="publish_point_cloud" default="true"/>

  <group ns="$(arg ns)">
    <node pkg="espros_cam660"
          type="cam660_node"
          name="$(arg nn)"
          output="screen"
          respawn="$(arg respawn)">


      <param name="lens_data" value="$(arg lens_data)"/>
      <param name="lens_center_offset_x" value="$(arg lens_center_offset_x)"/>
      <param name="lens_center_offset_y" value="$(arg lens_center_offset_y)"/>
      <param name="image_type" value="$(arg image_type)"/>
      <param name="trigger_single_shot" value="$(arg trigger_single_shot)"/>
      <param name="start_stream" value="$(arg start_stream)"/>
      <param name="hdr_mode" value="$(arg hdr_mode)"/>
      <param name="integration_time_tof_1" value="$(arg integration_time_tof_1)"/>
      <param name="integration_time_tof_2" value="$(arg integration_time_tof_2)"/>
      <param name="integration_time_tof_3" value="$(arg integration_time_tof_3)"/>
      <param name="integration_time_gray" value="$(arg integration_time_gray)"/>
      <param name="offset" value="$(arg offset)"/>
      <param name="roi_left_x"  value="$(arg roi_left_x)"/>
      <param name="roi_right_x" value="$(arg roi_right_x)"/>
      <param name="roi_height"  value="$(arg roi_height)"/>
      <param name="min_amplitude" value="$(arg min_amplitude)"/>
      <param name="min_distance" value="$(arg min_distance)"/>
      <param name="max_distance" value="$(arg max_distance)"/>
      <param name="frequence_modulation" value="$(arg frequence_modulation)"/>
      <param name="channel" value="$(arg channel)"/>
      <param name="median_filter" value="$(arg median_filter)"/>
      <param name="average_filter" value="$(arg average_filter)"/>
      <param name="edge_filter_threshold" value="$(arg edge_filter_threshold)"/>
      <param name="temporal_filter_factor" value="$(arg temporal_filter_factor)"/>
      <param name="temporal_filter_threshold" value="$(arg temporal_filter_threshold)"/>
      <param name="interference_detect_limit" value="$(arg interference_detect_limit)"/>
      <param name="use_last_value" value="$(arg use_last_value)"/>
      <param name="cartesian" value="$(arg cartesian)"/>
      <param name="publish_images" value="$(arg publish_images)"/>
      <param name="publish_point_cloud" value="$(arg publish_point_cloud)"/>

    </node>

    <node name="rviz" pkg="rviz" type="rviz" args= "-d $(find espros_cam660)/rviz/config_cam660.rviz" > </node>
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"> </node>

  </group>
</launch>
